aws/resource_aws_securityhub_organization_admin_account_test.go

```diff
@@ -0,0 +1,138 @@
+	resourceName := "aws_securityhub_organization_admin_account.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccOrganizationsAccountPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsSecurityHubOrganizationAdminAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSecurityHubOrganizationAdminAccountConfigSelf(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsSecurityHubOrganizationAdminAccountExists(resourceName),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsSecurityHubOrganizationAdminAccount(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsSecurityHubOrganizationAdminAccountDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).securityhubconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_securityhub_organization_admin_account" {
+			continue
+		}
+
+		adminAccount, err := finder.AdminAccount(conn, rs.Primary.ID)
+
+		// Because of this resource's dependencies, the Organizations organization and Security Hub account
+		// will be deleted first, resulting in the following valid error
+		if tfawserr.ErrMessageContains(err, securityhub.ErrCodeAccessDeniedException, "account is not a member of an organization") {
+			continue
+		}
+
+		if err != nil {
+			return err
+		}
+
+		if adminAccount == nil {
+			continue
+		}
+
+		return fmt.Errorf("expected Security Hub Organization Admin Account (%s) to be removed", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func testAccCheckAwsSecurityHubOrganizationAdminAccountExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).securityhubconn
+
+		adminAccount, err := finder.AdminAccount(conn, rs.Primary.ID)
+
+		if err != nil {
+			return err
+		}
+
+		if adminAccount == nil {
+			return fmt.Errorf("Security Hub Organization Admin Account (%s) not found", rs.Primary.ID)
+		}
+
+		return nil
+	}
+}
+
+func testAccSecurityHubOrganizationAdminAccountConfigSelf() string {
+	return `
+data "aws_caller_identity" "current" {}
+
+data "aws_partition" "current" {}
+
+resource "aws_organizations_organization" "test" {
+  aws_service_access_principals = ["securityhub.${data.aws_partition.current.dns_suffix}"]
+  feature_set                   = "ALL"
+}
+
+resource "aws_securityhub_account" "test" {}
+
+resource "aws_securityhub_organization_admin_account" "test" {
+  # Security Hub Account added as a dependency to avoid destroy-time error:
+  # InvalidInputException: Cannot disable Security Hub on the Security Hub administrator
+  depends_on = [aws_organizations_organization.test, aws_securityhub_account.test]
```

Comment:
alternatively we could catch the error `InvalidInputException: Cannot disable Security Hub on the Security Hub administrator1 in the `aws_securityhub_account` resource's _Delete_ method